src/main/java/api/kirana/register/customers/controllers/CustomersController.java

```diff
@@ -1,87 +0,0 @@
-package api.kirana.register.customers.controllers;
-
-import api.kirana.register.customers.models.CustomersDTO;
-import api.kirana.register.customers.service.CustomersServiceImpl;
-import api.kirana.register.response.ApiResponse;
-import jakarta.validation.Valid;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.data.domain.PageRequest;
-import org.springframework.data.domain.Pageable;
-import org.springframework.http.HttpStatus;
-import org.springframework.http.ResponseEntity;
-import org.springframework.web.bind.annotation.*;
-
-@RestController
-@RequestMapping("/api/customers")
-public class CustomersController {
-
-
-    private final CustomersServiceImpl customerService;
-
-    @Autowired
-    public CustomersController(CustomersServiceImpl customerService) {
-        this.customerService = customerService;
-    }
-
-    /**
-     * Retrieves a list of all customers.
-     *
-     * @return a ResponseEntity containing the list of all customers and HTTP status
-     */
-    @GetMapping("/all")
-    public ResponseEntity<ApiResponse> getAllCustomers(@RequestParam(defaultValue = "0") int page,
-                                                       @RequestParam(defaultValue = "2") int size) {
-        ApiResponse response = new ApiResponse();
-        Pageable pageable = PageRequest.of(page, size);
-        response.setData(customerService.getAllCustomers(pageable));
-        return new ResponseEntity<>(response, HttpStatus.OK);
-    }
-
-    /**
-     * Retrieves a customer by their ID.
-     *
-     * @param id the ID of the customer
-     * @return a ResponseEntity containing the customer data and HTTP status
-     */
-    @GetMapping()
-    public ResponseEntity<ApiResponse> getCustomer(
-           @Valid @RequestParam(required = false) String id,
-            @RequestParam(required = false) String name) {
-        ApiResponse response = new ApiResponse();
-        if (id == null && name == null) {
```

Comment:
Just add @NotNull annotation in request param instead of this null check, for example
@Valid  @NotNull @RequestParam(required = false) String id